pkg/pulumiyaml/run.go

```diff
@@ -1141,7 +1141,18 @@ func (e *programEvaluator) registerResource(kvp resourceNode) (lateboundResource
 		}
 	}
 	if v.Options.Import != nil {
-		opts = append(opts, pulumi.Import(pulumi.ID(v.Options.Import.Value)))
+		importValue, ok := e.evaluateExpr(v.Options.Import)
+		if ok {
+			if !hasOutputs(importValue) {
+				opts = append(opts, pulumi.Import(pulumi.ID(v.Options.Import.Value)))
+			} else {
+				e.error(v.Options.Import, "import must be not be an output")
+				overallOk = false
+			}
+		} else {
+			e.error(v.Options.Import, "couldn't evaluate the 'import' resource option")
+			overallOk = false
+		}
```

Comment:
This check should be done statically. This should not happen during runtime. 

Given the TODO in the PR, this code should be unreachable.

Reply:
That's true but I don't think the type checker today distinguishes between prompt/plain values and outputs. Does it?

Reply:
No. It would need to be a different pass then the type checker.